File: ICE3/ice3-resource/Q1/SoldierTest.java
/**
 * Created by yeowleong on 26/4/16.
 */
public class SoldierTest {
    public static void main(String[] args) {
        Commando c = new Commando("Peter", 5000, 200);
        Soldier s = new Soldier("John", 3000);

        System.out.println(c);
        System.out.println(s);

        System.out.println("Commando's base allowance : " + c.getBaseAllowance());
        System.out.println("Commando's gross allowance : " + c.getGrossAllowance());
        System.out.println("Soldier's base allowance : " + s.getBaseAllowance());
        System.out.println("Soldier's gross allowance : " + s.getGrossAllowance());

        // Commando{Soldier {name='Peter', baseAllowance=5000.0},
        // vocationAllowance=200.0}
        // Soldier {name='John', baseAllowance=3000.0}
        // Commando's base allowance is 5000.0
        // Commando's gross allowance is 5200.0

        // Soldier's base allowance is 3000.0
        // Soldier's gross allowance is 3000.0

    }
}

File: Solutions/ICE3-Solution/Q5/SoldierTest.py
from Commando import Commando
from Soldier import Soldier

if __name__ == '__main__':
    c = Commando("Peter", 5000, 200)
    s = Soldier("John", 3000)

    print(c)
    print(s)

    print("Commando's base allowance : {}".format(c.base_allowance))
    print("Commando's gross allowance : {}".format(c.gross_allowance))
    print("Soldier's base allowance : {}".format(s.base_allowance))
    print("Soldier's base allowance : {}".format(s.gross_allowance))
